clients/drcachesim/tools/view.cpp

```diff
@@ -258,7 +264,7 @@ view_t::parallel_shard_memref(void *shard_data, const memref_t &memref)
             // We delay printing until we know the tid.
             if (filetype_ == -1) {
                 filetype_ = static_cast<offline_file_type_t>(memref.marker.marker_value);
-                if (!init_from_filetype()) {
+                if (trace_version_ != -1 && !init_from_stream()) {
```

Comment:
Similar comments here: if we have both why do we need to init from stream?

How about splitting init_from_stream() into two pieces: one that gets filetype and version from the stream (could be called init_from_stream() or init_headers_from_stream_if_necessary() and is only called at 1st insr), and one that sets other state (probably called init_from_filetype()) and is called like today when we see the filetype or we see the first instr (and just called init_headers_from_stream_if_necessary())?